From a72c47c47c7d0a3f3cc03b3eb608489cd2061538 Mon Sep 17 00:00:00 2001
From: Przemyslaw Czarnota <przemyslawx.czarnota@intel.com>
Date: Thu, 17 Sep 2020 11:51:32 +0200
Subject: [PATCH] PONRTSYS-7968: Classify and trap OMCI packets to CPU

- Remove pon_qos_set_omci_tc()
- Always set the port and subinterface
---
 .../net/datapath/pon_qos/pon_qos_tc_ds_tc_remap.c  | 40 ----------------------
 drivers/net/datapath/pon_qos/pon_qos_tc_trap.c     | 12 +++----
 2 files changed, 5 insertions(+), 47 deletions(-)

diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_ds_tc_remap.c b/drivers/net/datapath/pon_qos/pon_qos_tc_ds_tc_remap.c
index 551a541dd0e5..57f39d1cedd5 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_ds_tc_remap.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_ds_tc_remap.c
@@ -352,42 +352,6 @@ int pon_qos_ev_tc_unassign(struct pon_qos_qdisc *sch, u8 tc)
 	return pon_qos_ev_assign_tc_upd(dev, dot1p, dot1p_sz);
 }
 
-/* TODO: remove this when CPU trap action is implemented */
-static int pon_qos_set_omci_tc(struct core_ops *gsw_handle)
-{
-	GSW_PCE_rule_t *rule;
-	GSW_return_t ret = 0;
-
-	if (!gsw_handle)
-		return -EINVAL;
-
-	/* Needed as rule struct is too large and triggers gcc warning */
-	rule = kzalloc(sizeof(*rule), GFP_KERNEL);
-	if (!rule)
-		return -ENOMEM;
-
-	/* TODO: Set ingress ds TC=8 for PON port subif=0 (OMCI) */
-	rule->pattern.nIndex = 0;
-	rule->pattern.bEnable = 1;
-	rule->pattern.bPortIdEnable = 1;
-	rule->pattern.nPortId = 2;
-	rule->pattern.bSubIfIdEnable = 1;
-	rule->pattern.nSubIfId = 0;
-
-	rule->action.eTrafficClassAction = 2;
-	rule->action.nTrafficClassAlternate = 8;
-	rule->action.ePortMapAction = 3;
-	rule->action.nForwardPortMap[0] = 1;
-
-	ret = gsw_handle->gsw_tflow_ops.TFLOW_PceRuleWrite(gsw_handle, rule);
-	if (ret != GSW_statusOk)
-		pr_err("%s: CPU ingress tc set failed: %d\n", __func__, ret);
-
-	kfree(rule);
-
-	return ret;
-}
-
 int pon_qos_tc_mappings_init(void)
 {
 	struct core_ops *gsw_handle = NULL;
@@ -433,9 +397,5 @@ int pon_qos_tc_mappings_init(void)
 		return ret;
 	}
 
-	ret = pon_qos_set_omci_tc(gsw_handle);
-	if (ret)
-		pr_err("%s: cannot set omci tc\n", __func__);
-
 	return ret;
 }
diff --git a/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c b/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c
index 7ec9336ed1f6..4aa556c19c40 100644
--- a/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c
+++ b/drivers/net/datapath/pon_qos/pon_qos_tc_trap.c
@@ -82,13 +82,11 @@ static int pon_qos_trap_set_sub_interface(struct net_device *dev,
 	 * for pon0.
 	 */
 	if (ret == DP_SUCCESS) {
-		if (dp_subif->port_id != 0) {
-			pce_rule->logicalportid = dp_subif->port_id;
-			pce_rule->pattern.bPortIdEnable = 1;
-			pce_rule->pattern.nPortId = dp_subif->port_id;
+		pce_rule->logicalportid = dp_subif->port_id;
+		pce_rule->pattern.bPortIdEnable = 1;
+		pce_rule->pattern.nPortId = dp_subif->port_id;
 
-			pce_blk_info->info.portid = dp_subif->port_id;
-		}
+		pce_blk_info->info.portid = dp_subif->port_id;
 
 		/* For pmapper set subifid based on bport */
 		if (dp_subif->flag_pmapper) {
@@ -100,7 +98,7 @@ static int pon_qos_trap_set_sub_interface(struct net_device *dev,
 			pce_rule->pattern.nSubIfId = dp_subif->bport;
 
 			pce_blk_info->info.subif = dp_subif->bport;
-		} else if (dp_subif->subif != 0) {
+		} else {
 			pce_rule->pattern.eSubIfIdType =
 				GSW_PCE_SUBIFID_TYPE_GROUP;
 			pce_rule->subifidgroup = dp_subif->subif_groupid;
